kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/VirtualKafkaClusterReconciler.java

```diff
@@ -127,27 +127,39 @@ public UpdateControl<VirtualKafkaCluster> reconcile(VirtualKafkaCluster cluster,
                     .filter(kpf -> hasAnyResolvedRefsFalse(Optional.ofNullable(kpf.getStatus()).map(KafkaProtocolFilterStatus::getConditions).orElse(List.of())))
                     .map(ResourcesUtil::toLocalRef)
                     .collect(Collectors.toCollection(TreeSet::new));
+
+            TreeSet<ProxyRef> ingressProxyRefs = existingIngresses.stream().map(i -> i.getSpec().getProxyRef()).collect(Collectors.toCollection(TreeSet::new));
+            // if the ingress references a different proxy to the virtual cluster, we will not have loaded the proxy.
+            // TODO can we discriminate between unresolved this mismatch?
+            TreeSet<ProxyRef> mismatchedIngressProxyRef = ingressProxyRefs.stream().filter(i -> !existingProxyRefs.contains(i))
+                    .collect(Collectors.toCollection(TreeSet::new));
+
             if (unresolvedServices.isEmpty()
                     && unresolvedIngresses.isEmpty()
-                    && unresolvedFilters.isEmpty()) {
-                conditions = context
+                    && unresolvedFilters.isEmpty()
+                    && mismatchedIngressProxyRef.isEmpty()) {
+                uc = context
                         .getSecondaryResource(ConfigMap.class, PROXY_CONFIG_MAP_EVENT_SOURCE_NAME)
                         .flatMap(cm -> Optional.ofNullable(cm.getData()))
                         .map(ProxyConfigData::new)
-                        .flatMap(data -> Optional.ofNullable(data.getConditionsForCluster(ResourcesUtil.name(cluster))))
-                        .orElse(List.of()); // cm not found, or this cluster missing in the data.
+                        .flatMap(data -> data.getStatusPatchForCluster(ResourcesUtil.name(cluster)))
+                        .map(patch -> {
+                            var rr = ResourceState.of(statusFactory.newTrueCondition(cluster, Condition.Type.ResolvedRefs));
+                            var acc = ResourceState.fromList(patch.getStatus().getConditions());
+                            return statusFactory.clusterStatusPatch(cluster, rr.replacementFor(acc));
```

Comment:
The precedence here feels funky to me. Asking RR to generate replacements for acc. Where acc is the existing state. To me it reads better (if logically equivalent) as `acc.updateWith(rr)` which would mean 
`statusFactory.clusterStatusPatch(cluster, acc.updatedWIth(rr))` would read as apply status patch to cluster using acc with updates from rr. 

But maybe thats just a me thing.

Reply:
yeah the language of how we combine the conditions is a bit tricky, we don't replace, we select "the freshest" by some algorithm. And then we need to also preserve the lastTransitionedTime if status hasn't changed. `updateWith` might be as good as any here, it's hard to name.